File: src/main/java/in/dragonbra/javasteam/steam/handlers/steamgameserver/LogOnDetails.java
package in.dragonbra.javasteam.steam.handlers.steamgameserver;

/**
 * Represents the details required to log into Steam3 as a game server.
 */
public class LogOnDetails {

    private String token;

    private int appID;

    /**
     * @return the authentication token used to log in as a game server
     */
    public String getToken() {
        return token;
    }

    /**
     * @param token the authentication token used to log in as a game server
     */
    public void setToken(String token) {
        this.token = token;
    }

    /**
     * @return the AppID this gameserver will serve
     */
    public int getAppID() {
        return appID;
    }

    /**
     * @param appID the AppID this gameserver will serve
     */
    public void setAppID(int appID) {
        this.appID = appID;
    }
}

File: src/main/java/in/dragonbra/javasteam/steam/handlers/steamgameserver/LogOnDetails.kt
package `in`.dragonbra.javasteam.steam.handlers.steamgameserver

/**
 * Represents the details required to log into Steam3 as a game server.
 *
 * @param token Gets or sets the authentication token used to log in as a game server.
 * @param appID Gets or sets the AppID this gameserver will serve.
 */
data class LogOnDetails(var token: String?, var appID: Int)
